plugins/module_utils/config_loader.py

```
@@ -97,8 +102,33 @@ def _parse_config_data(self, config_data):
         """Parses and assigns values from config_data."""
 
         # Assign token from config if it exists
-        if config_data["authentication"].get("token"):
-            setattr(self, "token", config_data["authentication"]["token"])
+        auth = config_data.get("authentication") or {}
+
+        token_to_use = auth.get("token-to-use")
+        access_token = auth.get("access-token")
+        refresh_token = auth.get("refresh-token")
+
+        if token_to_use == "refresh" and refresh_token:
```

Ruff (0.14.10)

Possible hardcoded password assigned to: "token_to_use"

(S105)